paddle/phi/kernels/gpu/multi_tensor_adam_kernel.cu

```diff
@@ -0,0 +1,278 @@
+        MT p = r_p[ii];
+        MT g = r_g[ii];
+        MT m = r_m[ii];
+        MT v = r_v[ii];
+        if (mode == false) {
+          m = beta1 * m + (static_cast<MT>(1.0) - beta1) * g;
+          v = beta2 * v + (static_cast<MT>(1.0) - beta2) * g * g;
+          r_m[ii] = m;
+          r_v[ii] = v;
+          MT denom =
+              (sqrt(v) / sqrt(static_cast<MT>(1.0) - beta2_pow)) + epsilon;
+          p += (m / denom) * (-(lr / (static_cast<MT>(1.0) - beta1_pow)));
+          r_p[ii] = p;
+        } else {  // weight decay
+          p *= (static_cast<MT>(1.0) - lr * decay);
+          m = beta1 * m + (static_cast<MT>(1.0) - beta1) * g;
+          v = beta2 * v + (static_cast<MT>(1.0) - beta2) * g * g;
+          r_m[ii] = m;
+          r_v[ii] = v;
+          MT denom =
+              (sqrt(v) / sqrt(static_cast<MT>(1.0) - beta2_pow)) + epsilon;
+          p += (m / denom) * (-(lr / (static_cast<MT>(1.0) - beta1_pow)));
+          r_p[ii] = p;
+        }
+      }
+#pragma unroll
+      for (int ii = 0; ii < PARALLEL_SIZE; ii++) {
+        int i = i_start + threadIdx.x + ii * blockDim.x;
+        if (i < n && i < compute_group_size) {
+          p[i] = static_cast<T>(r_p[ii]);
+          m[i] = r_m[ii];
+          v[i] = r_v[ii];
+          if (multi_precision) {
+            mp[i] = r_p[ii];
+          }
+        }
+      }
+    }
+  }
+};
+
+template <typename T>
+__global__ void UpdateBetaPow(T beta1,
+                              T beta2,
+                              const T* beta1_pow_,
+                              const T* beta2_pow_,
+                              T* beta1_pow_out,
+                              T* beta2_pow_out) {
+  *beta1_pow_out = beta1 * beta1_pow_[0];
+  *beta2_pow_out = beta2 * beta2_pow_[0];
+}
+
+template <typename T, typename Context>
+void MultiTensorAdamKernel(
+    const Context& dev_ctx,
+    const std::vector<const DenseTensor*>& params,
+    const std::vector<const DenseTensor*>& grads,
+    const std::vector<const DenseTensor*>& moments1,
+    const std::vector<const DenseTensor*>& moments2,
+    const paddle::optional<std::vector<const DenseTensor*>>& master_param,
+    const DenseTensor& beta1_pow,
+    const DenseTensor& beta2_pow,
+    const DenseTensor& learning_rate,
+    const paddle::optional<DenseTensor>& skip_update,
+    const Scalar& beta1,
+    const Scalar& beta2,
+    const Scalar& epsilon,
+    int compute_group_size,
+    float weight_decay,
+    bool mode,
+    bool multi_precision,
+    bool use_global_beta_pow,
+    std::vector<DenseTensor*> params_out,
+    std::vector<DenseTensor*> moments1_out,
+    std::vector<DenseTensor*> moments2_out,
+    std::vector<DenseTensor*> master_param_out,
+    DenseTensor* beta1_pow_out,
+    DenseTensor* beta2_pow_out) {
+  using MPDType = typename phi::dtype::MPTypeTrait<T>::Type;
+
+  VLOG(4) << "use_global_beta_pow:" << use_global_beta_pow;
+  MPDType beta1_ = beta1.to<MPDType>();
```

Comment:
Do not use `beta1_`. The suffix `_` means the data member in Google C++ coding style. https://google.github.io/styleguide/cppguide.html#Variable_Names